From 10ca73a69df5358dc59e5a02bc2486c5993b5726 Mon Sep 17 00:00:00 2001
From: David Holsgrove <david.holsgrove@xilinx.com>
Date: Wed, 27 Feb 2013 13:56:11 +1000
Subject: [PATCH 22/24] REVERT upstream change to garbage collection sweep which causes mb regression

Upstream change for PR13177 now clears the def_regular during gc_sweep of a
section. (All other archs in binutils/bfd/elf32-*.c received an update
to a warning about unresolvable relocations - this warning is not present
in binutils/bfd/elf32-microblaze.c, but this warning check would not
prevent the error being seen)

The visible issue with this change is when running a c++ application
in Petalinux which links libstdc++.so - it segfaults on execution.

This does not occur if static linking libstdc++.a, so its during the
relocations for a shared lib with garbage collection this occurs

Signed-off-by: David Holsgrove <david.holsgrove@xilinx.com>
---
 bfd/elflink.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/bfd/elflink.c b/bfd/elflink.c
index 50e65ef..cb945e8 100644
--- a/bfd/elflink.c
+++ b/bfd/elflink.c
@@ -11885,7 +11885,6 @@ elf_gc_sweep_symbol (struct elf_link_hash_entry *h, void *data)
 
       inf = (struct elf_gc_sweep_symbol_info *) data;
       (*inf->hide_symbol) (inf->info, h, TRUE);
-      h->def_regular = 0;
       h->ref_regular = 0;
       h->ref_regular_nonweak = 0;
     }
-- 
1.7.1

